src-web/components/modals/AnsibleAutomationModal.js

```diff
@@ -0,0 +1,387 @@
+    const { value } = event.currentTarget
+    this.setState({ansScheduleMode:value})
+  };
+
+  renderAnsibleScheduleMode(locale) {
+    const {ansScheduleMode} = this.state
+    return <React.Fragment>
+      <Title headingLevel="h2">
+        {msgs.get('modal.ansible.scheduleMode.title', locale)}
+      </Title>
+      <Text>
+        {msgs.get('modal.ansible.scheduleMode.info', locale)}
+      </Text>
+      <React.Fragment>
+        <Radio
+          isChecked={ansScheduleMode==='manual'}
+          name="manualRadio"
+          onChange={this.handleScheduleModeRadioChange}
+          label={msgs.get('modal.ansible.scheduleMode.manual.title', locale)}
+          id="manualRadio"
+          value="manual"
+          description={msgs.get('modal.ansible.scheduleMode.manual.info', locale)}
+        />
+        <Radio
+          isChecked={ansScheduleMode==='once'}
+          name="onceRadio"
+          onChange={this.handleScheduleModeRadioChange}
+          label={msgs.get('modal.ansible.scheduleMode.once.title', locale)}
+          id="onceRadio"
+          value="once"
+          description={msgs.get('modal.ansible.scheduleMode.once.info', locale)}
+        />
+        <Radio
+          isChecked={ansScheduleMode==='disable'}
+          name="disableRadio"
+          onChange={this.handleScheduleModeRadioChange}
+          label={msgs.get('modal.ansible.scheduleMode.disable.title', locale)}
+          id="disableRadio"
+          value="disable"
+          description={msgs.get('modal.ansible.scheduleMode.disable.info', locale)}
+        />
+      </React.Fragment>
+    </React.Fragment>
+  }
+
+  renderAnsibleHisotry(historyData) {
+    return <div>
+      {JSON.stringify(historyData)}
+    </div>
+  }
+}
+
+AnsibleAutomationModal.propTypes = {
+  data: PropTypes.shape({
+    metadata: PropTypes.object,
+    name: PropTypes.string,
+    namespace: PropTypes.string,
+  }),
+  handleClose: PropTypes.func,
+  // handleSubmit: PropTypes.func,
+  label: PropTypes.shape({
+    heading: PropTypes.string,
+    label: PropTypes.string,
+    primaryBtn: PropTypes.string,
+  }),
+  locale: PropTypes.string,
+  open:  PropTypes.bool,
+  reqErrorMsg:  PropTypes.string,
+  reqStatus:  PropTypes.string,
+  // resourceType: PropTypes.object,
+  type: PropTypes.string,
+}
+
+const mapStateToProps = state =>  {
+  return state.modal
+}
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    handleSubmit: (resourceType, namespace, name, data, resourceData, resourcePath, dispatchFun) => {
+      dispatch(dispatchFun(resourceType, namespace, name, data, resourceData, resourcePath))
+    },
+    handleClose: (modalType) => {
+      dispatch(clearRequestStatus())
+      dispatch(updateModal({open: false, type: modalType}))
+    },
+    receivePatchError: (resourceType, err) => {
+      dispatch(receivePatchError(err, resourceType))
+    }
+  }
+}
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AnsibleAutomationModal))
```

Comment:
`withRouter` doesn't seems to be needed as you are not using any props provided by it.